lib/clients.ex

```diff
@@ -6,33 +6,138 @@ defmodule GlassFactoryApi.Clients do
   alias GlassFactoryApi.Clients.Client
 
   @doc """
-  Get a client given the id
+  Return a tuple with `:ok` atom and the client from the specified ID
+  or an tuple with `:error` and a string with the error description.
 
   ## Examples
 
-      iex> GlassFactoryApi.Clients.get_client("1234")
-      %Client{
-        id: 1234,
-        name: "Google",
-        archived_at: null,
-        owner_id: 567,
-        office_id: 789
-      }
+  iex> GlassFactoryApi.Clients.get_client("1234")
+  {:ok,
+    %Client{
+      id: 1234,
+      name: "Google",
+      archived_at: null,
+      owner_id: 567,
+      office_id: 789
+    }
+  }
+
+
+  iex> GlassFactoryApi.Clients.get_client("1")
+  {:error, "Can't find a client with id 1"}
   """
   @spec get_client(integer(), map) :: {atom(), Client.t() | String.t()}
   def get_client(client_id, config \\ %{}) do
     with {:ok, %{status: 200, body: body}} <- ApiClient.get("clients/#{client_id}", config) do
       {:ok, Client.to_struct(body)}
     else
       {:ok, %{status: 404}} -> {:error, "Can't find a client with id #{client_id}"}
+      {:ok, %{body: body}} -> {:error, body}
+      {:error, error} -> {:error, error}
```

Comment:
Will Tesla guarantee `error` is a string here?

Reply:
@Cendrao Can you help me this one? 😊

Reply:
Tesla won't guarantee that, but I think `ApiClient` does, because of [this implementation](https://github.com/Cendrao/glass_factory_api/blob/c0498aa11ad70b0dccba80bf53a907c7ff8fa025/lib/api_client.ex#L21-L26).